redun/executors/k8s_utils.py

```diff
@@ -0,0 +1,108 @@
+# Utility routines for k8s API interactions
+# This library contains standardized routines for interacting with k8s through its API
+# It uses the Official Python client library for kubernetes:
+# https://github.com/kubernetes-client/python
+from typing import Dict
+
+import boto3
+from kubernetes import client, config
+
+from redun.executors.aws_utils import get_aws_env_vars
+
+DEFAULT_JOB_PREFIX = "redun-job"
+
+
+def get_k8s_batch_client():
+    """returns an API client supporting k8s batch API
+    https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/BatchV1Api.md"""
+    config.load_kube_config()
+    batch_v1 = client.BatchV1Api()
+    return batch_v1
+
+
+def get_k8s_core_client():
+    """returns an API client support k8s core API
+    https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/CoreV1Api.md"""
+    config.load_kube_config()
+    core_v1 = client.CoreV1Api()
+    return core_v1
+
+
+def create_resources(requests=None, limits=None):
+    """Creates resource limits for k8s pods
+    https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/V1ResourceRequirements.md"""
+    resources = client.V1ResourceRequirements()
+
+    if requests is None:
+        resources.requests = {}
+    else:
+        resources.requests = requests
+
+    if limits is None:
+        resources.limits = {}
+    else:
+        resources.limits = limits
+
+    return resources
+
+
+def create_job_object(
+    name=DEFAULT_JOB_PREFIX,
+    image="bash",
+    command="false",
+    resources=None,
+    timeout=None,
+    labels=None,
+    uid=None,
+    retries=1,
+):
+    """Creates a job object for redun job.
+    https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/V1Job.md
+    Also creates necessary sub-objects"""
+
+    env = [{"name": key, "value": value} for key, value in get_aws_env_vars().items()]
```

Comment:
Here, is where I propagate the AWS cred environment variables into the job Docker container.